### ISSUE TITLE
"expected a non-empty value" is a super ambiguous error message for empty strings

### ISSUE THREAD

**Issue by lforst** [getsentry/relay#2137]
I recommend we rephrase this to something like "expected a non-empty string". This is what it looks like in the interface:

![Screenshot 2023-05-19 at 15 12 10](https://github.com/getsentry/relay/assets/8118419/6cd5c815-0284-4a42-a319-083e2caca285)

Took me some JSON digging to understand what Sentry wants me to do.

https://github.com/getsentry/relay/blob/6f2946c479a44e96722c95ccf47b5985ce4e5dd5/relay-general/src/types/meta.rs#L314

**Comment by jjbayer** [comment]
@lforst knowing that release is a string, what additional information would you get by changing the wording from "value" to "string"?

**Comment by lforst** [comment]
Ok my reasoning may sound a bit weird but but "non-empty value" kinda sounds to me like it cannot be `null` or missing in general, which is not true. It can be missing.

"non-empty string" is more explicit in what the error message is actually about. Does that make sense?

**Comment by iker-barriocanal** [comment]
A `null` string is different from an empty string -- the string can be missing, but cannot be empty. I think this is fine, so I'll close the issue for now, and we can get back to it if the confusion is more generalized.

**Comment by lforst** [comment]
@iker-barriocanal I am not talking about a null string. A `""` causes an error saying "expected a non-empty value" which is hella confusing. Can we just rephrase this to "expected a non-empty string" it's a change that will take anybody 1 second.

**Comment by jjbayer** [comment]
> Can we just rephrase this to "expected a non-empty string" it's a change that will take anybody 1 second.

@lforst we're open to contributions!

**Comment by iker-barriocanal** [comment]
@lforst I don't think this change is that simple. We currently reuse that error in other places that may not be strings (see [here](https://github.com/getsentry/relay/blob/6bc8a240d00bd44afaff3f9f458ea5d51b882a4e/relay-general/src/store/schema.rs#L80)); I haven't tracked all the code but we may have non-string values checked too. Updating the message from "value" to "string" may mean other messages are now incorrect and generate more confusion. Updating this properly would require, at least, some investigation and differentiate string values from other values.

I decided to close the ticket because the amount of work here is bigger than it seems, and I believe the message is correct and not that confusing (I agree it could be clearer). As Joris suggests, PRs are welcome! The context above should help you get started.

Let me know if I missed something, or you think otherwise.
